sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1515,11 +1528,43 @@ def process(self, element, *schema_side_inputs):
 
     if not self.with_batched_input:
       row_and_insert_id = element[1]
+      row_byte_size = get_deep_size(row_and_insert_id)
+      # maintain buffer byte size for each destination
+      self._destination_buffer_byte_size.setdefault(destination, 0)
+      # send large rows that exceed BigQuery insert limits to DLQ
+      if row_byte_size >= self.max_insert_payload_size:
+        row_mb_size = row_byte_size / 1_000_000
+        max_mb_size = self.max_insert_payload_size / 1_000_000
+        error = (
```

Comment:
should these be added in _flush_batch? We should not return when one row exceeds the limit, right? We should put all the large rows as the failed rows.

Reply:
I overlooked this. but why does `_flush_batch` not check the row size?

Reply:
when `with_batched_input` is True, do we need to check this size limit?

Reply:
Yes that can trigger the same error if users batch their rows beforehand in a way that exceeds limits. However, I think handling this can be a separate effort as that is a different use case. This PR handles the more general use case of writing a PCollection of rows, which a few users were using when running into this issue

Reply:
Can we open one issue to track this?

Reply:
Opened #27363